plugins/modules/postgresql_pg_hba.py

```diff
@@ -351,32 +478,80 @@ def read(self):
         # read the pg_hbafile
         try:
             with open(self.pg_hba_file, 'r') as file:
-                for line in file:
-                    # split into line and comment
-                    line = line.strip()
-                    comment = None
-                    if '#' in line:
-                        line, comment = line.split('#', 1)
-                        if comment == '':
-                            comment = None
-                        line = line.rstrip()
-                    # if there is just a comment, save it
-                    if line == '':
-                        if comment is not None:
-                            self.comment.append('#' + comment)
-                    else:
-                        if comment is not None and not self.keep_comments_at_rules:
-                            # save the comment independent of the line
-                            self.comment.append('#' + comment)
-                            comment = None
-                        try:
-                            self.add_rule(PgHbaRule(line=line, comment=comment))
-                        except PgHbaRuleError:
-                            pass
-            self.unchanged()
-            self.preexisting_rules = dict(self.rules)
+                hba_string = file.read()
         except IOError:
-            pass
+            return
+
+        for line in parse_hba_file(hba_string):
+            if line["tokens"] == "COMMENT":
+                self.comment.append(line["comment"])
+            elif line["tokens"] != "EMPTY":
+                if not line["comment"]:
+                    self._from_tokens(line["tokens"])
+                else:
+                    if self.keep_comments_at_rules:
+                        self._from_tokens(line["tokens"], line["comment"])
+                    else:
+                        self.comment.append(line["comment"])
+                        self._from_tokens(line["tokens"])
+        self.unchanged()
+        self.preexisting_rules = dict(self.rules)
+
+    def _from_tokens(self, symbols, comment=None):
+        if len(symbols) < 4:
+            raise PgHbaRuleError("The rule has too few symbols")
```

Comment:
There are include rules which only consist of two symbols.
Maybe we don't need to check included files but I guess ignoring and keeping include rules is better than throwing an error.

Reply:
Yes, this is already planned for an upcoming PR. I have a lot more changes in the pipeline which I split up into several PRs on @Andersson007's request. I checked and this issue exists in the current version, so I would like to handle it in an upcoming PR, to avoid making this one larger than it needs to be.

Reply:
SGTM